compiler/rustc_codegen_llvm/src/debuginfo/metadata.rs

```diff
@@ -656,9 +656,12 @@ impl MsvcBasicName for ty::UintTy {
 
 impl MsvcBasicName for ty::FloatTy {
```

Comment:
Should this return something different since MSVC doesn't support these types?

Reply:
@compiler-errors do you know what to do here?

Reply:
@MaulingMonkey looks like you originally authored this function, do you know what needs to be done to support f16 and f128 here?

Reply:
Not sure if proper support is possible here without bugging Microsoft.  Even if it is possible, Microsoft hasn't bothered for their own SDK types (e.g. `_FLOAT128` can be found in the windows SDK, but it's just a struct around a couple of u64s and has no useful debug visualizers.)

If it *were* possible:
- Emit debug info for a `u16`/`u128`-wrapping structure (I'm fuzzy on the details here, I've only wrapped existing emitted types rather than emitting my own.  Merely defining an unused std type might not be enough as such types will get culled IME?)
- Write/extend a `*.natvis` file (probably [rust/src/etc/natvis/intrinsics.natvis](https://github.com/rust-lang/rust/blob/6bf600bc98879bf1bf2ab0f3b3b37c965f5bdff6/src/etc/natvis/intrinsic.natvis)?) that unpacks the bits and reconstitutes them as an approximate double or float in it's `<DisplayString>`.
  - How do you transmute/type-pun from `u64` → `f64` in the subset of C++ allowed by the debugger (no pointer-to-temporary shenannigans)?  *Maybe* abuse a union somehow...?
  - See [`natvis-pdbs` Recommended Reading](https://crates.io/crates/natvis-pdbs#recommended-reading)
  - If you added a new `*.nativs` file for some reason, make sure it's referenced properly (see e.g. [rust/src/bootstrap/src/core/build_steps/dist.rs](https://github.com/rust-lang/rust/blob/master/src/bootstrap/src/core/build_steps/dist.rs#L535-L538)
  - Consider unit tests like [tests/debuginfo/pretty-std-collections-hash.rs](https://github.com/rust-lang/rust/blob/6bf600bc98879bf1bf2ab0f3b3b37c965f5bdff6/tests/debuginfo/pretty-std-collections-hash.rs#L1)

Reply:
(It's [possible to call `_CrtCheckMemory()` from a watch window](https://ofekshilon.com/2016/11/01/checking-memory-corruption-from-the-debugger-in-2016/).  I forget if this works in a `*.natvis` visualizer.  If so, you could theoretically author `F128toApproxF64(high: u64, low: u64)` and stuff it in a DLL?  Or better yet, let someone more foolish take the time to go down *that* rabbit hole and shave that yak...)

Reply:
Thanks for the quick response! Is MsvcBasicName only used for debug info?

I am not very familiar with natvis so if this is noncritical, I will probably just leave it as a fixme until a follow up PR.

Reply:
Yep, this is just noncritical debug info.